### PR TITLE
philipye314 imp 11.5 Eligibility

### PR DESCRIPTION
**Pull request by philipye314**
### Description

[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
- Created EligibilityCard component on caseScreen
- Created EligibilityForm component that allows users to update their eligibility

[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy!"
IMG_4657
IMG_4656
IMG_4655

## How to review

[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'
- Add a new case and make sure you are getting the "CHECK ELIGIBILITY CARD" at first
- Update the eligibility to "YES I AM ELIGIBLE". Double check that you are now seeing the "FILE CLAIM" and "OPT OUT" cards.
- Do the same with a different case, update to "NO I AM NOT ELIGIBLE". Double check you will still get the "CHECK ELIGIBILITY" card.

## Next steps

[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."
- Need to fix typescript bug in cases.ts whoops
- Josh probably wants me to change the "FILE CLAIM" and "OPT OUT" buttons
- Temporary back button doesn't look very nice, doesn't make sense currently

### Online sources

[//]: # 'Optional - copy links to any tutorials or documentation that was useful to you when working on this PR'

### Related PRs

[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @stephaniewong2

### CODE REVIEW

File: src/app/(BottomTabNavigation)/Cases/CaseScreen/index.tsx [hunk 1]
```diff
@@ -8,12 +8,21 @@ import CaseSummaryCard from '../../../../Components/CaseSummaryCard/CaseSummarCa
 import EducationalBar from '../../../../Components/EducationalBar/EducationalBar';
 import EligibilityCard from '../../../../Components/EligibilityCard/EligibilityCard';
 import FormsCard from '../../../../Components/FormsCard/FormsCard';
-import { Case } from '../../../../types/types';
+import { Case, Eligibility } from '../../../../types/types';
+import { getCaseStatus } from '../../../../supabase/queries/cases';
 
 function CasesScreen() {
   const caseData = useLocalSearchParams() as unknown as Case;
+  const [status, setStatus] = useState(Eligibility.UNDETERMINED);
```
Comment (stephaniewong2) [inline src/app/(BottomTabNavigation)/Cases/CaseScreen/index.tsx]: Let's do `const [status, setStatus] = useState<Eligibility>();` instead! This fixes the issue of showing the screen for `UNDETERMINED` and `ELIGIBLE` in the case the user is eligible. We should also type all of our state variables!

File: src/app/(BottomTabNavigation)/Cases/CaseScreen/index.tsx [hunk 2]
```diff
@@ -8,12 +8,21 @@ import CaseSummaryCard from '../../../../Components/CaseSummaryCard/CaseSummarCa
...
 
-  console.log(caseData);
+  useEffect(() => {
+    const getStatus = async () => {
+      const caseStatus = await getCaseStatus(caseData.id);
+      setStatus(caseStatus);
+      console.log(caseStatus);
+    };
+    getStatus();
+  }, []);
```
Comment (stephaniewong2) [inline src/app/(BottomTabNavigation)/Cases/CaseScreen/index.tsx]: If we remove the dependency array, it will trigger every time we route to this page. This won't fix everything because we need to update our routing as well but don't worry about that because that needs to be fixed across the app!

File: src/app/(BottomTabNavigation)/Cases/CaseScreen/EligibilityForm/index.tsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+import { View, Text } from 'react-native';
+import { TouchableOpacity } from 'react-native-gesture-handler';
+import { router, useLocalSearchParams } from 'expo-router';
+import styles from './styles';
+import { updateCaseStatus } from '../../../../../supabase/queries/cases';
+import { CaseUid, Eligibility } from '../../../../../types/types';
+import Arrow from '../../../../../../assets/arrow.svg';
+
+export default function EligibilityForm() {
+  const { caseId } = useLocalSearchParams() as unknown as { caseId: CaseUid };
```
Comment (stephaniewong2) [inline src/app/(BottomTabNavigation)/Cases/CaseScreen/EligibilityForm/index.tsx:12]: I don't think we need the `as unknown` part here but keep it if it does not work without it!

File: src/app/(BottomTabNavigation)/Cases/CaseScreen/EligibilityForm/index.tsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
+import React, { useCallback, useEffect, useState } from 'react';
```
Comment (stephaniewong2) [inline src/app/(BottomTabNavigation)/Cases/CaseScreen/EligibilityForm/index.tsx]: Remove unused imports -- `useCallback, useEffect, useState`

File: src/Components/EligibilityCard/EligibilityCard.tsx [hunk 5]
```diff
@@ -1,10 +1,91 @@
 import React from 'react';
-import { View, Text } from 'react-native';
-
-export default function EligibilityCard() {
-  return (
-    <View>
-      <Text>EligibilityCard</Text>
-    </View>
-  );
+import { View, Text, Image } from 'react-native';
```
Comment (stephaniewong2) [inline src/Components/EligibilityCard/EligibilityCard.tsx]: Remove unused `Image` import. Could you also do `Cmd + S` to format imports

File: src/app/(BottomTabNavigation)/Cases/CaseScreen/EligibilityForm/index.tsx (hunk 4 above)
Comment (stephaniewong2) [reply src/app/(BottomTabNavigation)/Cases/CaseScreen/EligibilityForm/index.tsx]: Also run`Cmd + S` here to format imports

File: src/app/(BottomTabNavigation)/Cases/CaseScreen/EligibilityForm/styles.ts [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+    width: '100%',
+    height: '70%',
+    borderRadius: 5,
+    flexDirection: 'row',
+    justifyContent: 'space-between',
+    alignItems: 'center',
+    paddingHorizontal: 20,
+  },
+  buttonBottom: {
+    backgroundColor: '#2C2C2C',
```
Comment (stephaniewong2) [inline src/app/(BottomTabNavigation)/Cases/CaseScreen/EligibilityForm/styles.ts]: Let's add this color to the `colors` file and use it from there!

File: src/app/(BottomTabNavigation)/Cases/CaseScreen/EligibilityForm/styles.ts [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+  },
+  buttonTop: {
+    borderColor: 'black',
+    borderWidth: 0.5,
+  },
+  buttonBottomText: {
+    color: '#fff',
```
Comment (stephaniewong2) [inline src/app/(BottomTabNavigation)/Cases/CaseScreen/EligibilityForm/styles.ts]: Use color from `colors` file

File: src/app/(BottomTabNavigation)/Cases/CaseScreen/EligibilityForm/styles.ts [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+    flexDirection: 'row',
+    justifyContent: 'space-between',
+    alignItems: 'center',
+    paddingHorizontal: 20,
+  },
+  buttonBottom: {
+    backgroundColor: '#2C2C2C',
+  },
+  buttonTop: {
+    borderColor: 'black',
```
Comment (stephaniewong2) [inline src/app/(BottomTabNavigation)/Cases/CaseScreen/EligibilityForm/styles.ts]: Use `black` from `colors`

File: src/app/(BottomTabNavigation)/Cases/CaseScreen/EligibilityForm/styles.ts [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+  },
+  buttonBottomText: {
+    color: '#fff',
+  },
+  backButton: {
+    margin: 10,
+    alignItems: 'center',
+    backgroundColor: colors.midGrey,
+    padding: 10,
+    borderColor: 'black',
```
Comment (stephaniewong2) [inline src/app/(BottomTabNavigation)/Cases/CaseScreen/EligibilityForm/styles.ts]: Use `black` from `colors`

File: src/app/(BottomTabNavigation)/Cases/CaseScreen/index.tsx [hunk 10]
```diff
@@ -22,9 +31,15 @@ function CasesScreen() {
           <Text>Go Back</Text>
         </TouchableOpacity>
       </View>
+      {status === Eligibility.ELIGIBLE && (
```
Comment (stephaniewong2) [inline src/app/(BottomTabNavigation)/Cases/CaseScreen/index.tsx:35]: If eligible, the file claim and opt out buttons should lie below the `Case Status` bar and according to the Figma (new update), there should be a line with "OR" in between

**Review by stephaniewong2 — CHANGES_REQUESTED** [review]
Looks good! Just a few styling nits from me mostly! We also need to update routing for this to fully work but let's talk about that in person!

File: src/app/(BottomTabNavigation)/Cases/CaseScreen/index.tsx (hunk 10 above)
Comment (stephaniewong2) [reply src/app/(BottomTabNavigation)/Cases/CaseScreen/index.tsx:35]: Make sure you import the line as an SVG from Figma when you do this instead of hard coding a line with text!

**Review by stephaniewong2 — APPROVED** [review]
